lib/socketry/resolver.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Socketry
+  # DNS resolution subsystem
+  module Resolver
+    module_function
+
+    def addr(address)
+      return ::Resolv::IPv4.create(address)
```

Comment:
Isn't there a way to unify these IPvX calls without catching exceptions? The `ipaddr` stdlib could provide an instance which responds to `#ipv4?` and `#ipv6?`.